sparkle/platform/output.py

```diff
@@ -0,0 +1,298 @@
+            configurator: The configurator that was used
+            penalty_multiplier: penalty multiplier that is applied to the par performance
+            output: Path to the output directory
+        """
+        self.solver = resolve_object_name(solver_name,
+                                          gv.solver_nickname_mapping,
+                                          gv.solver_dir, Solver)
+        self.configurator = configurator
+        self.penalty_multiplier = penalty_multiplier
+        self.directory = path
+
+        if output is None:
+            output = path / "Analysis" / "configuration.json"
+        self.output = output
+
+        # TODO: Fix this spaghetti code to find the scenario file
+        solver_dir_name = path.name
+        scenario_dir = path / "outdir_train_configuration" / \
+            f"{solver_dir_name}_scenario" / "state-run0"
+        if scenario_dir.is_dir():
+            scenario_file = scenario_dir / "scenario.txt"
+            if not scenario_file.is_file():
+                print("Can't find scenario file")
+                return
+        else:
+            print("Can't find scenario file")
+            return
+
+        # Sets scenario on configurator object
+        self.initalise_scenario(scenario_file)
+        # Retrieves validation results and best configuration
+        self.training, self.best_config =\
+            self.get_validation_data(self.configurator.scenario.instance_set)
+        # Retrieve test validation results if they exist
+        if self.instance_set_test is not None:
+            self.test, _ = self.get_validation_data(self.instance_set_test)
+
+        self.write_output()
+
+    def initalise_scenario(self: ConfigurationOutput, scenario_file: Path) -> None:
+        """Reads scenario file and initalises ConfigurationScenario."""
+        config = {}
+        with scenario_file.open() as file:
+            for line in file:
+                key, value = line.strip().split(" = ")
+                config[key] = value
+
+        instance_set_train = resolve_object_name(
+            config["instance_file"],
+            gv.file_storage_data_mapping[gv.instances_nickname_path],
+            gv.instance_dir, InstanceSet)
+        # TODO: Find out why _train.txt is added to end of name
+        # Remove .txt to not cause any issues with later use of
+        # name in self.configurator.scenario._set_paths()
+        instance_set_train.name = instance_set_train.name.replace("_train.txt", "")
+
+        # Set instance_set_test to later retrieve validation results
+        instance_set_test = resolve_object_name(
+            config["test_instance_file"],
+            gv.file_storage_data_mapping[gv.instances_nickname_path],
+            gv.instance_dir, InstanceSet)
+        if instance_set_test is not None:
+            self.instance_set_test = instance_set_test
+
+        # Collect relevant settings
+        if "cpu_time" in config:
```

Comment:
Maybe these can be one liners (as we want to set a value in any case) like cpu_time = config["cpu_time"] if "cpu_time" in config else None